doc/ModulePriority.md

```diff
@@ -0,0 +1,146 @@
+# How to configure priorities for modules
+
+By default, IoT Edge does not impose an ordering in the sequence in which modules are started, updated or stopped. Edge Agent by default is the first module that gets started and based on the edge deployment specification, it figures out which modules need to be started, updated or stopped and executes those operations in a non-deterministic order.
+
+The processing order of modules can be controlled by specifying the value of a module-specific property called `priority` in the IoT Edge deployment. Modules that have been assigned a higher priority will be processed before modules that have been assigned a lower priority.
+
+**It must be noted that this feature is currently only supported for the in-memory database store of Edge Hub i.e., when the environment variable `UsePersistentStorage` is set to `false`. The work to support this feature for the disk-based database store is currently in progress.**
+
+## __Use case__
+
+Customers who have an array of modules of which some are 'critical' or 'foundation' modules that are required by other modules in the ecosystem might want these modules to be started before other modules. This is so as to achieve a better end user experience where other modules don't have to wait for these 'critical' or 'foundation' modules to be started, so as to initialize themselves.
+
+As an example, some customers want the Edge Hub module to be started before any other non-system modules in the ecosystem are started. This is so that other modules don't spend unnecessary cycles waiting for Edge Hub to come up before they can start sending messages to other modules or upstream to IoT Hub.
+
+**That being said, module owners should design their modules to withstand any failures of these 'critical' or 'foundation' modules, that they are dependent upon, as they could go down at any arbitrary time and an arbitrary number of times.**
+
+## __Configuration__
+
+Customers can optionally specify a `priority` value for each module in their IoT Edge deployment to achieve module boot ordering. Modules with a higher priority will be created and started first and only when an attempt has been made to start them, will other lower priority modules be created and started.
+
+The value of `priority` will be positive and zero-based with 0 being the highest priority value.
+A higher numeric `priority` value would indicate a lower priority assigned to that module. The maximum value of this property will be 4294967295.
```

Comment:
This is product feedback, not doc feedback, but this system of prioritization is backwards from the way IoT Edge prioritizes automatic deployments. It might cause confusion for customers when 1 is the lowest priority for deployments, but the highest priority for module within the deployment.

Reply:
I wasn’t are aware of the deployment prioritization design in Edge and I’d gone ahead and created a design spec (and subsequently implemented it) using the priority order starting from zero which is the highest priority.

The pull-request implementing this feature has been merged with the `master` branch of IoT Edge and I have notified our customers on how to use it with the priorities beginning from zero. Do we still wish to change this implementation to be more aligned with the deployment prioritization scheme?